Clarify test exception for JDK8 on OpenJ9

adamfarley opened the pull request:
This exception is specific to OpenJ9, so the exception should mention
it by name.

Signed-off-by: Adam Farley <adfarley@redhat.com>

adamfarley commented:
Closes https://github.com/adoptium/temurin-build/issues/2608

adamfarley commented:
Note: Ignoring the three types of linter failure (line too long, magic number, & TODO) for the same reason I ignored it here:

M-Davies reviewed (approved):
Looks good. Imo, it's easier to read this as the below but I won't block it on that account
```java
        if (jdkVersion.usesVM(VM.OPENJ9) && jdkVersion.isOlderOrEqual(9)) {
```

karianna commented on test/functional/buildAndPackage/src/net/adoptopenjdk/test/VendorPropertiesTest.java:
```diff
@@ -67,8 +67,8 @@ public VendorPropertiesTest() {
-        // Skip test on JDK8 for non-Hotspot JDKs.
-        if (!jdkVersion.isNewerOrEqual(9) && !jdkVersion.usesVM(VM.HOTSPOT)) {
+        // Skip test on JDK8 with OpenJ9 VM, as java -version output will not include a vendor name by design.
+        if (jdkVersion.usesVM(VM.OPENJ9) && !jdkVersion.isNewerOrEqual(9)) {
```
Prefer `if (jdkVersion.usesVM(VM.OPENJ9) && jdkVersion.isOlderThan(9)) {`

adamfarley replied:
Fixed, and I also tweaked the names of the other methods for consistency. Your thoughts?

llxia commented:
If the test is using TKG, then I think we should use playlist feature to limit JDK implementation and version. For example: 
https://github.com/adoptium/aqa-tests/blob/master/openjdk/playlist.xml#L467
https://github.com/adoptium/aqa-tests/blob/master/openjdk/playlist.xml#L476

smlambert commented:
Agree that is preferable (to use playlists to divide / exclude).  In order to do that, the tests will have to be refactored.

https://github.com/adoptium/temurin-build/blob/master/test/functional/buildAndPackage/src/net/adoptopenjdk/test/VendorPropertiesTest.java#L69-L98 would be moved to its own class and into its own testng and playlist target (outside of [CommonFeatureTests](https://github.com/adoptium/temurin-build/blob/master/test/functional/buildAndPackage/testng.xml#L18-L23) and outside of https://github.com/adoptium/temurin-build/blob/master/test/functional/buildAndPackage/playlist.xml#L49 so it can be limited to 11+.

karianna commented:
@adamfarley needs a rebase

adamfarley commented:
@karianna - Rebased. 

Also I removed the superfluous changes to isNewerOrEqual and isNewerOrEqualSameFeature because it made the method names unnecessarily long.

@sophia-guo and @smlambert - I agree that an exclude should happen at the playlist level.

On inspection, the rebase would require 
- the VmPropertiesChecks interface (along with implimentations) to be broken out, or made externally accessible.
- there to be two entries in the playlist for one test, in order to enable both OpenJ9 (11+) testing and Hotspot (8+) testing.

Preliminary work towards this end seemed to add complexity, so I had a rethink and moved the logic inside AdoptiumPropertiesChecks. That way, instead of skipping the test, we explicitly test for the different behaviour on OpenJ9, JDK8. We do this in other tests, explicitly testing for different behaviour, and it seemed to simplify the whole business.

What do you think?

smlambert reviewed (approved):
Approving this approach as a 'temporary' measure.  As discussed in previous comments, we should ideally restructure the smoke tests and leverage the playlist filtering for jdkVersion checks, rather than expect each type of test to manage its own checking (which then requires us to maintain code for version checking in multiple places).  

Ultimately, a follow-up refactoring PR will be preferential and help simplify and reduce tech debt.
